src/backend/distributed/planner/relation_restriction_equivalence.c

```diff
@@ -2149,6 +2150,16 @@ GetRestrictInfoListForRelation(RangeTblEntry *rangeTblEntry,
 			continue;
 		}
 
+		/*
+		 * PlaceHolderVar is not expected in a restriction clause. Otherwise,
+		 * pull_var_clause_default would throw error. PG would create physical
+		 * restriction anyway, so it is safe to skip this restriction.
+		 */
+		if (FindNodeMatchingCheckFunction((Node *) restrictionClause, HasPlaceHolderVar))
```

Comment:
I find the reading of the comment a little hard. We first say that `PlaceHolderVar is not expected in a restriction clause.` but we also try to skip PlaceHolderVar from a restrict clause.  

PlaceHolderVars are not relevant for processing the restrictInfos or such